homeassistant/components/zwave_js/sensor.py

```diff
@@ -547,13 +595,14 @@ def __init__(
         unit_of_measurement: str | None = None,
     ) -> None:
         """Initialize a ZWaveSensorBase entity."""
-        super().__init__(config_entry, driver, info)
         self.entity_description = entity_description
+        super().__init__(config_entry, driver, info)
         self._attr_native_unit_of_measurement = unit_of_measurement
 
         # Entity class attributes
         self._attr_force_update = True
-        self._attr_name = self.generate_name(include_value_name=True)
+        if not entity_description.name:
```

Comment:
This looks like it would affect more entities than the ones we add here.

Reply:
I checked, it would not. We only use entity description names in sensors. The default names for the Energy Production CC values are IMO too vague

Reply:
What about all the other sensors that inherit this class?

Reply:
none of them currently use entity description names. The only other entity description names we use are for the statistics sensors which are not from a subclass of this base class

Reply:
Ok.